Added some missing offsets for dedicated server.

nukeulater opened the pull request:
Fixed some compiler warnings.
Reorganized some code to follow the new module format (check Camera, UI).
Moved all startup hooks to the appropriate file.
Renamed teamspeak class pointers to be more distinguishable. 
Disabled custom engine initalization until ready.

nukeulater committed:
Disabled custom engine initialization until ready

Added a few more offsets for dedicated server.

nukeulater committed:
Renamed teamspeak class pointers to be more distinguishable.

nukeulater committed:
Moved initialization hooks to corresponding file.

nukeulater committed:
Added check I forgot to add.

nukeulater committed:
Removed test stuff.

nukeulater committed:
Removed unused patch

nukeulater committed:
Game globals*

nukeulater committed:
Moved UI stuff to Modules/UI...

and cleaned up.

nukeulater committed:
Cleaned H2Tweaks

setCrosshairSize too (still not good enough but better).

nukeulater committed:
Added missing offsets to get_device_acceleration_scale

nukeulater committed:
Moved camera and crosshair stuff in their own H2MOD module files.

Fixed some warnings.
Removed more unused include files for various cpp files.

nukeulater committed:
Got rid of the developer account requirement to access live list menus

PermaNulled commented:
We didn't want to get into the game engine after the game type had loaded we needed an in between point where if we called the original function it would load everything (similar to on death pre death), honestly getting tired of people changing that hook. The whole deal was supposed to be on map load.

nukeulater committed:
Revert name change on hook.

NukeULater commented:
I renamed it back (the offsets are the same).

nukeulater committed:
Removed comments.

iSupersniper commented:
Why did you change this?

https://github.com/PermaNulled/cartographer/pull/150/files#diff-6b9d191874e4d21a1826e25cbb45cfa2R1133

iSupersniper commented:
I believe Permanull restructured the code to be on a module base, not a category base.  So instead of having 1 centralized cpp file for all stuff modifying the mainmenu, we have a folder named main menu with each modification it's own module.  Therefore any code that is related to it can be easily found by people who are new to the project and want to help out.  This creates an SDK environment.

NukeULater commented:
Most of the stuff inside Modules folder is Glitchy's / Yoshi's code, w/ renamed file names and put in folders, categorised by the function they do.
I did the same by creating a folder with all the UI related stuff, based on what element they modify (ex radar, crosshair) and centralize it within the UI header file that has a namespace where we place functions/variables, which is very friendly to work with imo.

NukeULater commented:
To avoid any further issues/confusion, I sugggest you guys write down some code structure rules to follow.

nukeulater committed:
Fixed BlamLib SetBit problem

nukeulater committed:
Moved RawInput in it's own folder and made Input header for input related stuff

Updated gitignore to avoid *.user files.
Removed some unused includes and code.

pnill committed:
Merge branch 'development' into offsets

nukeulater committed:
Merge remote-tracking branch 'origin/development' into offsets

nukeulater committed:
Re-enable player actions update rate patch

nukeulater committed:
Quick fix for XEnumerate thread spamming

Not the best way but it'll work for now.

nukeulater committed:
Fixed single player crash when touching a device

nukeulater committed:
Better fix for radar (now works in single-player too)

nukeulater committed:
Disable unnecessary hook for now

shayaantx commented on xlive/Globals.cpp:
```diff
@@ -25,9 +25,9 @@ Players* players = new Players();
-TSClient* client = NULL;
-TSServer* server = NULL;
+IN_ADDR clientMachineAddress = { };
+TSClient* TS_client = NULL;
```
why...this naming is completely different with naming of every other variable in this file?

nukeulater replied:
im sure the new naming is better than "server" and "client"

shayaantx commented on xlive/H2MOD.cpp:
```diff
@@ -1912,16 +1881,20 @@ VOID CALLBACK UpdateDiscordStateTimer(HWND hwnd, UINT uMsg, UINT_PTR idEvent, DW
 		this->Server = TRUE;
+
+		// Initialize teamspeak server
+		if (TS_server == NULL) 
```
last I checked this happens when we apply hooks for the dedi? has that changed now?

shayaantx commented on xlive/H2MOD/Modules/Camera/FOV/FOV.cpp:
```diff
@@ -0,0 +1,25 @@
+	{
+		float current_FOV = *reinterpret_cast<float*>(h2mod->GetBase() + 0x41D984);
+
+		//int res_width = *(int*)(H2BaseAddr + 0xA3DA00); //wip
```
if this code isn't ready, is this method ready for commit?

if you have stuff not ready, it should be prefixed with a TODO/issue/feature number

shayaantx commented:
please submit PR's as we discussed, this is too massive of a change
